test/deprecated.jl

```diff
@@ -1,6 +1,73 @@
+# Define some rules to test One on
+dummy_identity(x) = x
+@scalar_rule(dummy_identity(x), One())
+
+very_nice(x, y) = x + y
+@scalar_rule(very_nice(x, y), (One(), One()))
```

Comment:
this same function name is used with a different definition in `test/rules.jl`, which is confusing -- can we rename this here?